Library/Homebrew/cask/cmd/automerge.rb

```diff
@@ -108,7 +112,10 @@ def diff_only_version_or_checksum_changed(diff)
         return false if additions.count != deletions.count
         return false if additions.count > 2
 
-        changed_lines.all? { |line| diff_line_is_version(line.to_s) || diff_line_is_sha256(line.to_s) }
+        changed_lines.all? do |line|
+          (diff_line_is_version(line.to_s) && check_allowed_chars(line.to_s)) ||
```

Comment:
Might want to use a `next if` and `next unless` here as it's not super easy to parse when this will pass/fail using binary precedence.